### ISSUE TITLE
Doesn't work with Plex

### ISSUE THREAD

**Issue by turkeyphant** [folletto/Blipshot#26]
Keeps requesting to "reload the page". Blipshot still doesn't work immediately after reloading the page.

![image](https://user-images.githubusercontent.com/8129583/147416406-b91b73ca-94a5-4425-9f29-05b139df3b72.png)

**Comment by folletto** [comment]
Can you please share the link to the page that doesn't work?

**Comment by turkeyphant** [comment]
> Can you please share the link to the page that doesn't work?

I cannot. It's a local server.

**Comment by folletto** [comment]
Ah ok, that might be the problem. I think Chrome has a different access level for locally loaded pages, and probably won't allow it at all for local files (not served via a HTTP server). Can you still (in anonymized form if necessary) paste the URL?

**Comment by turkeyphant** [comment]
It is served by at HTTP server. It's just not on the public internet.

The URL is https://192.168.0.100:32400 so it makes no difference if I give it to you or not.

**Comment by folletto** [comment]
Thanks. I don't know Plex so I have no idea how it loads local files.

Thanks for the URL too. I didn't ask because I wanted to access, I asked because it provides informations on how the file is served (https, numeric ip, special port, no sub-folder).

I'll have a look.

**Comment by turkeyphant** [comment]
Sorry that redirects to http://IP:32400/web/index.html

Some pages are served like this:

http://IP:32400/web/index.html#!/media/d4e31e86faee098031f50e9b6dd3cf34e99b9c00/com.plexapp.plugins.library?source=playlists

**Comment by folletto** [comment]
Ok I checked the Chrome limitations, and it seems it should work (the block is only for `file` access). I ruled out that option.

I then installed Plex locally, and it seems working... However... it doesn't scroll. 

![pageshot of 'Plex' @ 2021-12-26-1845'50](https://user-images.githubusercontent.com/4389/147417451-cf1cb6c7-c735-4727-9081-cfc2c0a9601d.png)

As far as I dug into it, Plex scrolling is done by adding `overflow-y: scroll` to a `div` instead of having the whole page scroll, which means the plugin as it is built today is unable to get that scrolling, as it's not the full page scrolling, but just a piece of it.
I'm not even sure it's possible for any plugin to make that work, because even the Developer Console "Capture full size screenshot" is not able to scroll it. 

<img width="561" alt="Screenshot 2021-12-26 at 18 53 05" src="https://user-images.githubusercontent.com/4389/147417435-2f88eeb6-5f80-4118-a841-c268fe4f25b5.png">

I'm sorry I don't have better news.

**Comment by turkeyphant** [comment]
Thanks for this.

The issue I am having doesn't seem related to `overflow-y: scroll` styling. As per my screenshot, it won't even start and it gives a misleading error message.

However, if it's not possible without adapting the plugin to scroll div elements that's also good to know.

**Comment by folletto** [comment]
Yes the issue you mention is separate, it doesn't seem to be related to that: I wasn't able to identify exactly what's triggering it.

For background, that error message usually happens when the page-side JS from the plugin isn't injected correctly (it tries to call the function, and the function doesn't exist): that's why it suggests a reload. My guess is that there's something on that page that might be rewriting the content in some way and either disrupting the plugin JS or blocking it from loading entirely.

I just noted the `overflow-y: scroll` issue to say that _even if_ I was able to fix that issue, given Plex is not scrolling a page normally but using that CSS feature to scroll part of the page, it still wouldn't work. 

The plugin can't easily adapted to scroll that because inherently `overflow: scroll` behaviours would scroll just a chunk of a page, and there might be multiple on the page, so even if it was possible to select the one to scroll, how would it deal with the rest of the page? (like, in a universal way).

That said, as a workaround, maybe that specific page is different and all... you can try to use that Developer Console feature and see if it works in your case?
https://www.howtogeek.com/423558/how-to-take-full-page-screenshots-in-google-chrome-without-using-an-extension/

**Comment by turkeyphant** [comment]
Unfortunately that just takes a screenshot of the currently-viewable area.

**Comment by folletto** [comment]
Them I'm afraid it won't be possible to screenshot that full page without rewriting the specific page CSS. 
Sorry about that.

**Comment by turkeyphant** [comment]
If I used inspect element to change the CSS or to remove other parts of the page leaving on the main `<div>` would that help?

**Comment by folletto** [comment]
Yup. If you can edit the CSS / HTML to create a normal scrolling page it should.
